Fix for maxRetries=0

stickeegreg opened the pull request:
Fixes #1095

brettmc commented on src/SDK/Common/Export/Http/PsrTransport.php:
```diff
@@ -92,7 +92,7 @@ public function send(string $payload, ?CancellationInterface $cancellation = nul
         }
 
-        for ($retries = 0;; $retries++) {
+        for ($tries = 0;; $tries++) {
```
I don't think the variable name change is needed.

brettmc reviewed (changes requested):
There's a failing test with this change. Can you look into why, and make that green? I assume the problem here is that if max retire = 0 then it never tries. Does the spec say anything about valid values for maxRetries?

stickeegreg commented:
Thanks, I've fixed the test and added a new one, also undone renaming the variable.

I had a look at the spec, and as far as I can see it doesn't mention a retry limit at all.
Looking at the Go implementation it seems like they use a maximum time rather than number of tries.

brettmc commented on src/SDK/Common/Export/Http/PsrTransport.php:
```diff
@@ -110,7 +110,7 @@ public function send(string $payload, ?CancellationInterface $cancellation = nul
             }
 
-            if ($retries + 1 === $this->maxRetries) {
+            if ($retries === $this->maxRetries) {
```
what happens if retries is negative? perhaps this should be a `>=` test...

stickeegreg replied:
Tempted to say "you get what you deserve" if you pass a negative maxRetries 😂 
I've pushed up a change so it's `>=` now.
